### PR TITLE
[Redis] Add `--redis-version` param to `az redis create` and `az redis update`

### PR DESCRIPTION
**Pull request by MECHANDR**
**Description**
Upgrade azure-mgmt-redis library version from 12.0.0 to 13.0.0
Add `--redis-version` param to `az redis create`  and `az redis update` commands

**Testing Guide**
`az redis create -n {name} -g {rg} -l {location} --sku {sku} --vm-size {size} --redis-version {redis_version}`

**History Notes**
Add `--redis-version` param to `az redis create`  and `az redis update` commands

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Redis

File: src/azure-cli/HISTORY.rst [hunk 1]
```diff
@@ -20,6 +20,10 @@ Release History
 * Hotfix: `az storage fs directory download`: Fix the issue with `--sas-token` to generate valid sas url (#18811)
 * Hotfix: `az storage blob copy start`: Fix the issue in copy from different account (#18730)
 
+**Redis**
+
+*Add param `--redis-version` to `az redis create` and `az redis update` commands
+
```
Comment (evelyn-ys) [inline src/azure-cli/HISTORY.rst:26]: No need to modify `HISTORY.rst` manually.
It will be auto-generated by release pipeline according to PR title

Comment (MECHANDR) [reply src/azure-cli/HISTORY.rst]: Reverted changes

File: src/azure-cli/azure/cli/command_modules/redis/tests/latest/test_redis_scenario.py [hunk 2]
```diff
@@ -86,11 +86,33 @@ def test_redis_cache_with_tlsversion_and_settings(self, resource_group):
...
+        self.kwargs = {
+            'rg': resource_group,
+            'name': self.create_random_name(prefix=name_prefix, length=24),
+            'location': location,
+            'sku': basic_sku,
+            'size': basic_size,
+            'redis_version': '6'
+        }
+
+        self.cmd('az redis create -n {name} -g {rg} -l {location} --sku {sku} --vm-size {size} --redis-version {redis_version}')
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/redis/tests/latest/test_redis_scenario.py:101]: Can you add tests for updating redis version too?

Comment (MECHANDR) [reply src/azure-cli/azure/cli/command_modules/redis/tests/latest/test_redis_scenario.py:101]: Passing `None` will not cause any updates as it not valid. Update will happen only if valid value is passed which is currently 4 or 6.

Comment (evelyn-ys) [reply src/azure-cli/azure/cli/command_modules/redis/tests/latest/test_redis_scenario.py:101]: Then can you just add tests for updating. Like to version 4

File: src/azure-cli/azure/cli/command_modules/redis/_params.py [hunk 3]
```diff
@@ -41,7 +41,8 @@ def load_arguments(self, _):
         c.argument('minimum_tls_version', help='Specifies the TLS version required by clients to connect to cache', arg_type=get_enum_type(TlsVersion))
         c.argument('vm_size', arg_type=get_enum_type(allowed_c_family_sizes + allowed_p_family_sizes), help='Size of Redis cache to deploy. Basic and Standard Cache sizes start with C. Premium Cache sizes start with P')
         c.argument('enable_non_ssl_port', action='store_true', help='If specified, then the non-ssl redis server port (6379) will be enabled.')
-        c.argument('replicas_per_master', help='The number of replicas to be created per master.', is_preview=True)
+        c.argument('replicas_per_master', help='The number of replicas to be created per master.')
+        c.argument('redis_version', help='Redis version. Only major version will be used in create/update request with current valid values: (4, 6)')
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/redis/_params.py:45]: Can you add validator here to check if the given value is 4 or 6?

Comment (MECHANDR) [reply src/azure-cli/azure/cli/command_modules/redis/_params.py:45]: Will prefer to do this validation in resource provider.

File: src/azure-cli/azure/cli/command_modules/redis/tests/latest/test_redis_scenario.py (hunk 2 above)
Comment (MECHANDR) [reply src/azure-cli/azure/cli/command_modules/redis/tests/latest/test_redis_scenario.py:101]: updating redis version work is in progress and currently it doesn't change the resource redis version, so didn't include update test case

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**